synchronizer/synchronizer.go

```diff
@@ -315,6 +333,20 @@ func (s *Synchronizer) AdvanceView(syncInfo hotstuff.SyncInfo) {
 		}
 	}
 
+	b, ok := syncInfo.Block()
```

Comment:
Not sure the `ok` return thing adds much value; isn't this just as clear?
```go
if b := syncInfo.Block(); b != nil {
	s.updateLeafBlock(b)
}
```

Reply:
I guess that is the same for QC and TC. Check the following:
```go
// TC returns the timeout certificate, if present.
func (si SyncInfo) TC() (_ TimeoutCert, _ bool) {
	if si.tc != nil {
		return *si.tc, true
	}
	return
}

// AggQC returns the AggregateQC, if present.
func (si SyncInfo) AggQC() (_ AggregateQC, _ bool) {
	if si.aggQC != nil {
		return *si.aggQC, true
	}
	return
}

// Block returns the block, if present.
func (si SyncInfo) Block() (_ *Block, _ bool) {
	if si.block != nil {
		return si.block, true
	}
	return
}
```